src/dns/coredns.rs

```diff
@@ -428,7 +428,11 @@ async fn forward_dns_req(mut cl: AsyncClient, message: Message) -> Option<Messag
 
     match cl.send(req).try_next().await {
         Ok(Some(mut response)) => {
-            response.set_id(id);
+            let mut res = response.clone().into_message();
+            res.set_id(id);
+            if let Ok(resp) = DnsResponse::from_message(res) {
+                response = resp
+            }
```

Comment:
This looks backwards? Why convert to Message juts to convert back to DnsResponse to then again convert back to Message in the Some() statement?

To me it should be more like this to avoid the extra conversions:
```
Ok(Some(response)) => {
            for answer in response.answers() {
                debug!(
                    "{} {} {} {} => {:#?}",
                    id,
                    answer.name().to_string(),
                    answer.record_type(),
                    answer.dns_class(),
                    answer.data(),
                );
            }
            let mut new_message = response.into_message();
            new_message.set_id(id);
            Some(new_message)
        }
```

Reply:
SGTM